crates/api/api_utils/src/utils.rs

```diff
@@ -73,15 +72,15 @@ pub async fn check_is_mod_or_admin(
   pool: &mut DbPool<'_>,
   person_id: PersonId,
   community_id: CommunityId,
-  local_instance_id: InstanceId,
 ) -> LemmyResult<()> {
-  let is_mod =
-    CommunityModeratorView::check_is_community_moderator(pool, community_id, person_id).await;
-  if is_mod.is_ok()
-    || PersonView::read(pool, person_id, None, local_instance_id, false)
-      .await
-      .is_ok_and(|t| t.is_admin)
-  {
+  let is_mod = CommunityModeratorView::check_is_community_moderator(pool, community_id, person_id)
+    .await
+    .is_ok();
+  let is_admin = LocalUserView::read_person(pool, person_id)
+    .await
+    .is_ok_and(|t| t.local_user.admin);
+
+  if is_mod || is_admin {
```

Comment:
The failing `deletePost` API tests were caused by logical mistakes in this file. It should've been reading admin status from the localuserview, not the `personview`.